### ISSUE TITLE
[x86-64] [Linux] Game crash on Steam Linux Runtime

### ISSUE THREAD

**Issue by ryu-ketsueki** [Facepunch/garrysmod-issues#5092]
### Details
I recently came across a problem that just happened to be the same of another issue. It was the one of an openSUSE user who couldn't open the game because the game couldn't load a specific library. The explanation given was that it would seem the library is not the version the game expects. Having said that, Steam Linux Runtime does provide the libraries the game needs in the correct version. And it is what happened. The warning box did not appear and the game seemed to load just fine, until it simply closed with no warning box saying what could have went wrong. I would like to know what can be done in this situation to try debug this problem and find a solution.

Just so you know, Gmod works perfectly under Proton, although with a much slower loading speed. It is that difference in loading that made me try the native port again, although the Lua Panic Not Enough Memory error still pop up once the map finish loading. I've read online that the x86-64 branch was meant to solve that and it is why I'm testing it.

### Steps to reproduce
Just switch to x86-64 branch and enable Steam Linux Runtime in the compatibility layer

**Comment by Synkstar** [comment]
Yeah it seems after the OCT 25th update(#12946355) the 64 bit version broke in windows and linux. Including the server. If I have my server in 64 bit it crashes. Also some of my binary modules do not load. I tried disabling them though to see if that was whats causing the crash on my server but it still crashed. Then I realized the client is broken for my friend on windows and me. I really shouldn't have auto update on my server 2bh now that I think about it. For the meantime the old chromium branch loads fine.

**Comment by ryu-ketsueki** [comment]
Chromium branch got the same issue for me. Can't get past the loading screen. And as of right now, Gmod is really unstable, even on the Windows version through proton, occasionally showing this error below which sometimes gets fixed on its own after reinstalling and sometimes don't, being a total coin toss as for if I'll ever be able to play the game again, so if the Linux port can be made playable again, would help a lot.
![image](https://user-images.githubusercontent.com/39316730/140582876-38fa9470-586e-4ec7-929d-b6d68eb1f1be.png)

**Comment by ryu-ketsueki** [comment]
By the way, this is what appears in the txt for the crash on the chromium branch:
```
-Lua Stack Traces-
==================
  Client
    Lua Interface = NULL

  Server
    Lua Interface = NULL

  MenuSystem
    Lua Interface = NULL

-Console Buffer-
================
1(8.241872):  Attempting to load Chromium...
```
I can send the .dmp file if required

**Comment by robotboy655** [comment]
I have tested GMod on `Ubuntu 20.04 LTS` with and without Steam Runtime enabled, both worked perfectly.

**Comment by ryu-ketsueki** [comment]
> I have tested GMod on `Ubuntu 20.04 LTS` with and without Steam Runtime enabled, both worked perfectly.

Was the memory leak fixed then? I couldn't test it yet on my end

**Comment by keepofthehause** [comment]
> I have tested GMod on `Ubuntu 20.04 LTS` with and without Steam Runtime enabled, both worked perfectly.

Issue now occurs on Arch Linux after the toolchain update.

**Comment by TibixDev** [comment]
Having the same problem on Manjaro, just after the update.

**Comment by Levev** [comment]
After updating to the latest Manjaro version, the same issue occurs to me too.

**Comment by TibixDev** [comment]
**Update**

Using [this](https://github.com/solsticegamestudios/GModCEFCodecFix/) fixed it for me.

**Comment by Blazzy12** [comment]
> **Update**
> 
> Using [this](https://github.com/solsticegamestudios/GModCEFCodecFix/) fixed it for me.

It happened to me too (Manjaro) this works as a fix ty so much Tibix!

**Comment by ipaqmaster** [comment]
> **Update**
> 
> Using [this](https://github.com/solsticegamestudios/GModCEFCodecFix/) fixed it for me.

This also did it for me. Thanks, couldn't reach the main menu before on Arch.

**Comment by Xandertron** [comment]
> **Update**
> 
> Using [this](https://github.com/solsticegamestudios/GModCEFCodecFix/) fixed it for me.

also worked for ubuntu 22.04 (or whatever current version it is)

**Comment by forkiesassds** [comment]
> > **Update**
> > Using [this](https://github.com/solsticegamestudios/GModCEFCodecFix/) fixed it for me.
> 
> also worked for ubuntu 22.04 (or whatever current version it is)

Doesn't work for me, crashes during connection screen when starting singleplayer or multiplayer. Even without any addons.

**Comment by Cameron-Stuart** [comment]
> **Update**
> 
> Using [this](https://github.com/solsticegamestudios/GModCEFCodecFix/) fixed it for me.

Are you still using the Steam Runtime, or are you using Steam Native (most likely through the steam-native package on Manjaro/Arch)? I'm trying to figure out why even with the CEF fix I get a segfault on startup with no logs under the x86-64 branch (and the chromium branch) on Fedora 36 using the Steam Runtime

**Comment by dinkley** [comment]
> **Update**
> 
> Using [this](https://github.com/solsticegamestudios/GModCEFCodecFix/) fixed it for me.

Confirmed on Arch Linux 6.17.3-zen2-1-zen kernel. Game would crash on "Starting serverside" before running the patch tool. Now it launches properly. If I get VAC banned because of this I'm going to be fuming LMAO

Specs if its helpful for the devs:
OS: Arch Linux x86_64
Kernel: Linux 6.17.3-zen2-1-zen
CPU: AMD Ryzen 7 5800X
GPU: AMD Radeon RX 5600 XT
Memory: 32 GiB
